src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java

```diff
@@ -365,4 +367,59 @@ public void setRegistrationDatesOnGroups(final List<IliasNode> groups, final Loc
             }
         }
     }
+
+    /**
+     * Adds a new group to a given course.
+     * If the group already exists, the user will be asked, if he
+     * still wants to create it. Unfortunately the exist mechanism searches in all
+     * courses of the user and not just in the given one. We cannot restrict this search, thus
+     * the user has to decide. (@see {@link com.github.mavogel.ilias.lib.wrapper.IliasEndpoint})
+     *
+     * @param course    The course the group shall be created in
+     * @param groupName The name of the group
+     * @return True iff the group has been successfully been created
+     * @throws Exception in case of a failure. Detailed logs are written.
+     */
+    @Override
+    public boolean addGroup(IliasNode course, String groupName) throws Exception {
```

Comment:
I'd suggest 2 Methods though if it is not possible to determine in which course the group already exists

1. Check if a group already exists
```java
public boolean groupExists(String groupName)
```
2. Adds a group within the given course. Throws an exception if it fails
```java
public void addGroup(IliasNode course, String groupName) throws Exception
```

Then handle the User IO in the calling client, like e.g. the `IliasClient` or your library

Reply:
Yes that makes more sense. Done too.